addons/l10n_in_edi/models/account_edi_format.py

```diff
@@ -65,6 +67,11 @@ def _check_move_configuration(self, move):
         if not re.match("^.{1,16}$", move.name):
             error_message.append(_("Invoice number should not be more than 16 characters"))
         for line in move.invoice_line_ids.filtered(lambda line: not (line.display_type or line.is_rounding_line)):
+            if line.price_subtotal < 0:
+                # Line having a negative amount is not allowed.
+                if not move._l10n_in_edi_is_managing_invoice_negative_lines_allowed():
+                    raise ValidationError(_("Invoice lines having a negative amount are not allowed to generate the IRN. "
+                                  "Please create a credit note instead."))
```

Comment:
this is not normal that we merged a negative line with a positive one.
because it creates diff what we have in odoo and what we send to the government.

so before active this, our consultant told him how the negative line is handled.
we improve this in future with documentation and put this in the setting.